src/main/resources/templates/userProfile.html

```diff
@@ -11,6 +11,6 @@
 <h1>Welcome to Your Personal Page</h1>
 <p th:text="'Hello, ' + ${session.loggedInUserName}"></p>
 
-<a th:href="logOut">Logout</a>
+<a th:href="@{'logOut?page=' + session.activePage}">Logout</a>
```

Comment:
I added a page parameter to the logout, so that when the user logs out they can still see what page they are on in the URL.